src/Features/CSharp/Portable/SignatureHelp/AttributeSignatureHelpProvider.cs

```diff
@@ -94,10 +92,10 @@ protected override async Task<SignatureHelpItems> GetItemsWorkerAsync(Document d
                 return null;
             }
 
-            var anonymousTypeDisplayService = document.GetLanguageService<IAnonymousTypeDisplayService>();
-            var documentationCommentFormatter = document.GetLanguageService<IDocumentationCommentFormattingService>();
-            var textSpan = SignatureHelpUtilities.GetSignatureHelpSpan(attribute.ArgumentList);
-            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var anonymousTypeDisplayService = document.GetRequiredLanguageService<IAnonymousTypeDisplayService>();
+            var documentationCommentFormatter = document.GetRequiredLanguageService<IDocumentationCommentFormattingService>();
+            var textSpan = SignatureHelpUtilities.GetSignatureHelpSpan(attribute.ArgumentList!);
```

Comment:
super nit: Instead of using `!` here and in other areas, would it potentially be more readable if Contract checks were used instead? It would also allow us to throw sooner if there's an issue. It's probably personal preference though, and I also don't know if there's a fundamental difference between the two approaches I'm missing.

Reply:
That's a good point. I've been using `!` mostly, guess I will be using `Contract.ThrowIfNull` going forward :)